.travis.yml

```diff
@@ -39,54 +63,88 @@ addons:
+    if: 'env(VALIDATE_TOKENS) = yes'
+  - name: check
+    if: 'NOT env(VALIDATE_TOKENS) = yes'
+  - name: spec
+    if: 'NOT env(VALIDATE_TOKENS) = yes'
+  - name: deploy
+    if: 'tag IS present AND NOT env(VALIDATE_TOKENS) = yes'
 
+jobs:
   include:
     - stage: check
       name: 'Syntax, style, and validation checks'
-      rvm: 2.4.5
+      rvm: 2.4.9
       env: PUPPET_VERSION="~> 5"
       script:
         - bundle exec rake pkg:compare_latest_tag
         - bundle exec rake pkg:create_tag_changelog
+
     - stage: spec
-      name: 'Puppet 5.3 (PE 2017.3)'
-      rvm: 2.4.4
-      env: PUPPET_VERSION="~> 5.3.0"
+      rvm: 2.4.9
+      name: 'Puppet 5.5 (SIMP 6.4, PE 2018.1)'
+      env: PUPPET_VERSION="~> 5.5.0"
       script:
         - bundle exec rake spec
 
     - stage: spec
-      name: 'Latest Puppet 5.x'
-      rvm: 2.4.5
+      name: 'Puppet 5.x (Latest)'
+      rvm: 2.4.9
       env: PUPPET_VERSION="~> 5.0"
       script:
         - bundle exec rake spec
 
     - stage: spec
-      name: 'Latest Puppet 6.x (allowed to fail)'
-      rvm: 2.5.1
-      env: PUPPET_VERSION="~> 6.0"
+      name: 'Puppet 6.10 (PE 2019.2)'
+      rvm: 2.5.7
+      env: PUPPET_VERSION="~> 6.10.0"
       script:
         - bundle exec rake spec
 
     - stage: deploy
-      rvm: 2.4.5
+      rvm: 2.4.9
+      env: PUPPET_VERSION="~> 5.5.0"
       script:
         - true
       before_deploy:
-        - "export SPECFILE_VERSION=`rpm -q --specfile build/*.spec --queryformat '%{VERSION}\n' | head -1`"
-        - '[[ $TRAVIS_TAG =~ ^${SPECFILE_VERSION}$ ]]'
+        - "export PUPMOD_METADATA_VERSION=`ruby -r json -e \"puts JSON.parse(File.read('metadata.json')).fetch('version')\"`"
+        - '[[ $TRAVIS_TAG =~ ^simp-${PUPMOD_METADATA_VERSION}$|^${PUPMOD_METADATA_VERSION}$ ]]'
+        - 'gem install -v "~> 5.5.0" puppet'
+        - 'git clean -f -x -d'
+        - 'puppet module build'
+        - 'find pkg -name ''*.tar.gz'''
       deploy:
-        - provider: releases
-          api_key:
-              secure: "iDqlp0aodfNFl0ItK1jf07hN+z9EsuK39loAN/qfXfs/0bnWIByiAHwEefOQoBxzEVdjoFmnOw36SG3qGTgXwbOWQQE3s17yh6S6J6uLMt5WKcuUVL3cNgUtngeam8Eva/beC7BG2kqQB6rIVJpp0iodlhmbT5afnCiJUuNhR8LZdf8DDAs7fulwUmi/mS9Pv2DthLE0BdPBa7iGHI4UxQz0B/evyCKvDCx7BQSYy3NQl4uBA4eGhuJF8EhazBXK0Px82RRqIsKWOkFvP3fOgxBtvf5r3BL/2rzpuEDFjGfbVvmaAvQQyShGSCDoz01h97bcASqQiYrHO6DRTH219I6TLFPyENXtFRIUBFRE7SAw95md81mNoSVsasXtUbghz4HJspOqAYXGJ0zT/lUqVRbQq62/fkOe9+Y5B4eYRTKKxia5J8+pvS33Vadld8f0o2i858bmpaaX7tk0tzLKdtJ44o7ORnIeNxoa1Z9c+QsGvjVFRDLPzWTbcCugECIlRNKHrecTBrA9bOVMtx/03ZaPcFWhMTESHy2tKKlUdYfQT4CgUifklzvm8gcSeZ8TOoKylD9i3EkztJUi9eMFx6h5Kshf6yNfTzmZYoY8dFvRjjBiRhtC3SeY+miwBsOYUq5pchRgkKV1kxENEpecRwL8wySJmzZ+x89IKQMwKT8="
+        - provider: script
           skip_cleanup: true
+          script: 'curl -sS --fail -A "$FORGE_USER_AGENT" -H "Authorization: Bearer ${PUPPETFORGE_API_TOKEN}" -X POST -F "file=@$(find $PWD/pkg -name ''*.tar.gz'')" https://forgeapi.puppet.com/v3/releases'
```

Comment:
Adapter doesn't publish to the Forge so something isn't right here.

Reply:
I didn't customize properly.  Will mark WIP and fix.